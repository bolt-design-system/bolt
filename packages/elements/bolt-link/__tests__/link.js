import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../testing/testing-helpers';
import schema from '../link.schema';
const { display, valign } = schema.properties;

const timeout = 90000;

describe('link', () => {
  let page;

  afterAll(async () => {
    await stopServer();
    await page.close();
  });

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  test('basic link', async () => {
    const results = await render('@bolt-elements-link/link.twig', {
      text: 'Hello World',
      url: 'https://pega.com',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
