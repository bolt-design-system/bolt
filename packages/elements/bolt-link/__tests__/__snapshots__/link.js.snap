// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link basic link 1`] = `
<bolt-link display="inline"
           valign="center"
           url="https://pega.com"
>
  <a href="https://pega.com"
     is="shadow-root"
     class="c-bolt-link  c-bolt-link--display-inline c-bolt-link--valign-center"
  >
    <ssr-keep for="bolt-link"
              class="c-bolt-link__text"
    >
      Hello World
    </ssr-keep>
  </a>
</bolt-link>
`;
